src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -569,6 +581,15 @@ private static ReturnCode displaySummary(int fileCount, int invalidCount, int sk
         }
     }
 
+    /**
+     * Tells the validator to monitor progress and show ongoing validation progress
+     *
+     * @param id an identifier for the progress monitor
+     */
+    public void setProgressMonitor(String id) {
```

Comment:
As we create a new Monitor for each run, monitorID is really a local variable for a single validation run. This makes sense as we create a new file each time we create a new monitor. This should probably be passed into `validateKBAndReturnReport` rather than set as state for the object. Alternatively, we could create the Monitor object here and use that object all runs the validator performs.

Reply:
I'll go with B), to create the monitor object in `setProgressMonitor()`.